src/WappMatchers/UpLevelMatcherTrait.php

```diff
@@ -15,7 +16,11 @@ abstract protected function doMatch(Filesystem $fs, string $path): array;
     public function match(Filesystem $fs, string $path): array
     {
         if (!$result = $this->doMatch($fs, $path)) {
-            return $this->doMatch($fs, rtrim($path) . '/../');
+            try {
+                $result = $this->doMatch($fs, rtrim($path) . '/../');
```

Comment:
any chance to match $path against $basePath to prevent unnecessary lookup?

Reply:
no, `$fs` provides no access to basePath.

also the problem can be for cases like 
```
$basePath = '/';
$scanDir = '/inaccessible/www-root`;
```
so `/inaccessble` will be inside our basePath, but still throw exception.